fe/fe-core/src/main/java/com/starrocks/sql/optimizer/ChildOutputPropertyGuarantor.java

```diff
@@ -137,11 +137,37 @@ private void enforceChildShuffleDistribution(List<Integer> shuffleColumns, Group
         childrenOutputProperties.set(childIndex, newChildInputProperty);
     }
 
-    private void transToBucketShuffleJoin(HashDistributionDesc leftLocalDistributionDesc,
+    private void transToBucketShuffleJoin(HashDistributionSpec leftLocalDistributionSpec,
                                           List<Integer> leftShuffleColumns, List<Integer> rightShuffleColumns) {
         List<Integer> bucketShuffleColumns = Lists.newArrayList();
+        HashDistributionDesc leftLocalDistributionDesc = leftLocalDistributionSpec.getHashDistributionDesc();
         for (int leftScanColumn : leftLocalDistributionDesc.getColumns()) {
             int index = leftShuffleColumns.indexOf(leftScanColumn);
+            if (index == -1) {
```

Comment:
Why need to add this ？it looks wired

Reply:
The local distribution columns must in the required shuffle columns，otherwise the local property could not satify the shuffle required property